File: tests/smoke/fastapi-sqlalchemy/app/repositories/order_repository.py
from sqlalchemy import select
from app.models.order import Order


class OrderRepository:
    def find_by_id(self, order_id: int):
        # 실제론 session.scalar(select(Order).where(...))
        return None

    def insert(self, dto: dict):
        order = Order(**dto)
        return order

File: tests/smoke/nestjs-prisma/src/order/order.repository.ts
import { Injectable } from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

@Injectable()
export class OrderRepository {
  async findById(id: number) {
    return prisma.order.findUnique({ where: { id } });
  }

  async insert(dto: Record<string, unknown>) {
    return prisma.order.create({ data: dto as any });
  }
}
